### PR TITLE
WEBAPP-550: Disable events if flag is not set

### PR DESCRIPTION
**Pull request by Schedulaar**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/modules/app/route-configs/EventsRouteConfig.js [hunk 1]
```diff
@@ -52,16 +54,20 @@ class EventsRouteConfig implements RouteConfig<EventsRouteParamsType, RequiredPa
...
-  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({ events: payloads.eventsPayload })
+  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({
+    cities: payloads.citiesPayload,
+    events: payloads.eventsPayload
+  })
 
-  getPageTitle = ({ t, payloads, cityName, location }) => {
-    if (!cityName) {
+  getPageTitle = ({ t, payloads, location }) => {
+    const city = payloads.cities.data && payloads.cities.data.find(_city => _city.code === location.payload.city)
```
Comment (klinzo) [inline src/modules/app/route-configs/EventsRouteConfig.js:63]: I really hate that we have to do this checks all the time...

Comment (klinzo) [reply src/modules/app/route-configs/EventsRouteConfig.js:63]: Maybe we could refactor this by adding the current city model to the location or something like that...

File: src/modules/app/route-configs/EventsRouteConfig.js [hunk 2]
```diff
@@ -52,16 +54,20 @@ class EventsRouteConfig implements RouteConfig<EventsRouteParamsType, RequiredPa
...
 
-  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({ events: payloads.eventsPayload })
+  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({
+    cities: payloads.citiesPayload,
+    events: payloads.eventsPayload
+  })
 
-  getPageTitle = ({ t, payloads, cityName, location }) => {
-    if (!cityName) {
+  getPageTitle = ({ t, payloads, location }) => {
```
Comment (klinzo) [inline src/modules/app/route-configs/EventsRouteConfig.js:62]: The cityName isn't passed to this function? What about the other routes? Why does it work for e.g. `CategoriesRoute`?

File: src/modules/app/route-configs/ExtrasRouteConfig.js [hunk 3]
```diff
@@ -47,12 +48,23 @@ class ExtrasRouteConfig implements RouteConfig<ExtrasRouteParamsType, RequiredPa
...
+  })
 
   getLanguageChangePath = ({ location, language }) =>
     this.getRoutePath({ city: location.payload.city, language })
 
-  getPageTitle = ({ t, cityName }) => cityName ? `${t('pageTitles.extras')} - ${cityName}` : null
+  getPageTitle = ({ t, payloads, location }) => {
+    const cityModel = payloads.cities.data
+      ? payloads.cities.data.find(cityModel => cityModel.code === location.payload.city)
+      : null
```
Comment (klinzo) [inline src/modules/app/route-configs/ExtrasRouteConfig.js:62]:
```suggestion
    const cityModel = payloads.cities.data
      && payloads.cities.data.find(cityModel => cityModel.code === location.payload.city)
```

File: src/modules/app/route-configs/ExtrasRouteConfig.js [hunk 4]
```diff
@@ -47,12 +48,23 @@ class ExtrasRouteConfig implements RouteConfig<ExtrasRouteParamsType, RequiredPa
...
+    if (!cityModel || !cityModel.extrasEnabled) {
+      return null
```
Comment (klinzo) [inline src/modules/app/route-configs/ExtrasRouteConfig.js:64]: This means the page name defaults to `Integreat`, right?

File: src/modules/app/route-configs/EventsRouteConfig.js (hunk 1 above)
Comment (klinzo) [reply src/modules/app/route-configs/EventsRouteConfig.js:63]: Of course not in the scope of this PR

File: src/modules/app/route-configs/EventsRouteConfig.js (hunk 2 above)
Comment (Schedulaar) [reply src/modules/app/route-configs/EventsRouteConfig.js:62]: Oh yes, it is, but as I need the complete cityModel anyways, I decided not to take it from there

File: src/modules/app/route-configs/EventsRouteConfig.js (hunk 1 above)
Comment (Schedulaar) [reply src/modules/app/route-configs/EventsRouteConfig.js:63]: Yeah, we could think about replacing `cityName: ?string` with `cityModel: ?CityModel` in `getPageTitle`

File: src/modules/app/route-configs/ExtrasRouteConfig.js (hunk 4 above)
Comment (Schedulaar) [reply src/modules/app/route-configs/ExtrasRouteConfig.js:64]: Yes

**Comment by Schedulaar** [comment]
@klinzo What changes did you request? Afaics, your comments are only slight codestyle preferences.

**Pull request merged**